### ISSUE TITLE
[Feature Request] If a player only has 1 home available, teleport him to this home on /home

### ISSUE THREAD

**Issue by Daniil1288** [RestoreMonarchyPlugins/MoreHomes#1]
I'd like to request the following feature.
For example, I have my Default homes set to 1 like so: <DefaultHomes>1</DefaultHomes>
So when a player tries to teleport to his home using /home he still needs to specify which home to teleport to using /home <name> which is a little counterintuitive.
I suggest that the plugin checks whether the player only has one home available and if he does, it should teleport him to that specific home without asking for its name. So that players can use /home without parameters when they don't have additional homes.

Also, I don't think that /home or /homes should be available to console... /renamehome has this set to player-only, which is correct, though.
[01.08.2019 14:27:40] [Error] An error occured while executing home []: ```
System.InvalidCastException: Specified cast is not valid.
  at MoreHomes.Commands.CommandHome.Execute (Rocket.API.IRocketPlayer caller, System.String[] command) [0x00014] in <e2a7cc3e66f9422bb3ef4ac425e8f72c>:0 
  at Rocket.Core.Commands.RocketCommandManager+RegisteredRocketCommand.Execute (Rocket.API.IRocketPlayer caller, System.String[] command) [0x00000] in <e897f45c5c654e3ea071255fe5aed0d5>:0 
  at Rocket.Core.Commands.RocketCommandManager.Execute (Rocket.API.IRocketPlayer player, System.String command) [0x00176] in <e897f45c5c654e3ea071255fe5aed0d5>:0 
```

```
[01.08.2019 14:33:29] [Error] An error occured while executing homes []: System.InvalidCastException: Specified cast is not valid.
  at MoreHomes.Commands.CommandHomes.Execute (Rocket.API.IRocketPlayer caller, System.String[] command) [0x00000] in <e2a7cc3e66f9422bb3ef4ac425e8f72c>:0 
  at Rocket.Core.Commands.RocketCommandManager+RegisteredRocketCommand.Execute (Rocket.API.IRocketPlayer caller, System.String[] command) [0x00000] in <e897f45c5c654e3ea071255fe5aed0d5>:0 
  at Rocket.Core.Commands.RocketCommandManager.Execute (Rocket.API.IRocketPlayer player, System.String command) [0x00176] in <e897f45c5c654e3ea071255fe5aed0d5>:0 
```

**Comment by Daniil1288** [comment]
Also, had some players experience a weird bug that /restorehomes couldn't fix.
Here are the screenshots provided to me by them (two people experiencing the same thing): 
1. /homes tells the player that he has 1 home available.
![image](https://user-images.githubusercontent.com/1404955/62291437-a0dfcd00-b46c-11e9-8fc1-055e3d85863f.png)
2. Plugin errors out.
![image](https://user-images.githubusercontent.com/1404955/62291511-c79e0380-b46c-11e9-8027-a78e562d5c68.png)
Not sure how to resolve this without wiping the database.

**Comment by RestoreMonarchy** [comment]
Hey @Daniil1288 

1. I'll probably add this feature within this or next week, however what do you think of having a UI for that, so if you type /home and have more than one bed it opens the UI and you click on one of the beds. I can easily make this one too and UI can be enable/disabled (cause probably non-workshop server won't want to use it).

2. Thank you for reporting that I've just changed it.
> Also, I don't think that /home or /homes should be available to console... /renamehome has this set to player-only, which is correct, though. 

3. About the players' experience the error, I've messed up with translations, because it should have replied that the bed was destroyed while delay instead of that it couldn't be found. Fixed it in the latest patch too.

4. I'm probably gonna add some command like `/destroybed` or `/removehome` along with the UI feature.

Greetings,
MCrow

**Comment by Daniil1288** [comment]
@RestoreMonarchy,
I agree that the UI feature is a great addition to modded servers but as I also have a few without mods I'd like it to be optional. 

I see that you've changed the `/restorehomes` command's AllowedCaller to Player. Could we have Both? That is, if an admin is not in-game but has the console connection open he doesn't have to boot the game up and go to every single server just to execute the command.

I feel that you could add both `/destroybed` and `/removehome`: one would be for players (`/destroybed <home_name>`) and one for admins (`/removehome <player_name> <home_name>`). That would also require to include some way for an admin to know which beds the players has (maybe by adding some separate permission to `/homes` so that it could be executed as `/homes <player_name>`) so it's up to you. Oh, and both commands should check whether the bed actually exists in-game and if it does, it should also be destroyed.

**Comment by RestoreMonarchy** [comment]
Hi @Daniil1288 

I changed `/restorehomes` AllowedCaller to Both just now. About what you've said about `/destroybed`, `/removehome` and `/homes <playerName>` I'll probably do so along with the UI feature when I'll be doing one.

**Comment by Daniil1288** [comment]
@RestoreMonarchy, Hey, 
Noticed that you've made some changes on the august-changes branch. 
Just wanted to let you know that I tested that version of the plugin out and currently it doesn't work with this error:
[28.10.2019 20:57:08] [Info] [World] [Owner] mrDra'vin [mrDra'vin]: "/home"
[28.10.2019 20:57:39] [Error] An error occured while executing home []: System.NullReferenceException: Object reference not set to an instance of an object
  at MoreHomes.Helpers.Utilities.ShouldAllowTeleport (MoreHomes.Models.PlayerBed bed) [0x0000e] in <2b7ca0ed9d5844d896f83b685c34667f>:0 
  at MoreHomes.Commands.CommandHome.Execute (Rocket.API.IRocketPlayer caller, System.String[] command) [0x000de] in <2b7ca0ed9d5844d896f83b685c34667f>:0 
  at Rocket.Core.Commands.RocketCommandManager+RegisteredRocketCommand.Execute (Rocket.API.IRocketPlayer caller, System.String[] command) [0x00000] in <466fa14d466749559a64a11ae8acbc60>:0 
  at Rocket.Core.Commands.RocketCommandManager.Execute (Rocket.API.IRocketPlayer player, System.String command) [0x001b2] in <466fa14d466749559a64a11ae8acbc60>:0 
Thanks.

**Comment by Daniil1288** [comment]
@RestoreMonarchy hi again, about morehomes, still having to do /restorehomes every restart otherwise the beds won't show up for players in /homes any way to fix this? ![image](https://user-images.githubusercontent.com/1404955/75354591-469dd200-58be-11ea-8ae7-c57a31183e54.png)

**Comment by Daniil1288** [comment]
Also a player has just reported to me that for some reason when he uses /home the command doesn't always work and it sometimes doesn't even teleport him (using latest version of the Teleportation plugin too). Here's a video that he had sent me: https://drive.google.com/file/d/1YT_bH5YVIHRIxzKlNHzoj-oF80o0w2yV/view?usp=sharing

**Comment by RestoreMonarchy** [comment]
Try out the 1.2.2.2 version and let me know if it works https://github.com/RestoreMonarchyPlugins/MoreHomes/releases/tag/1.2.2.2
